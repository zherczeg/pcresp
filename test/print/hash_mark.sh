#!/bin/bash

CWD=`pwd`
if [ -f "../../bin/pcresp" ]; then
    PATH="$CWD/../../bin":$PATH
else
  if [ -f "../bin/pcresp" ]; then
      PATH="$CWD/../bin":$PATH
  else
      echo "Cannot find pcresp build directory"
      exit
  fi
fi

echo "echo A | pcresp '(.)' -s '*print    AB    CD   <AB  CD>    < AB >     CD'"
echo A | pcresp '(.)' -s '*print    AB    CD   <AB  CD>    < AB >     CD'
echo

echo "A | pcresp '(*:MarK)(.)' -s '*print #1 #{1} #< #> ## #M'"
echo A | pcresp '(*:MarK)(.)' -s '*print #1 #{1} #< #> ## #M'
echo

echo "A | pcresp -d key value -d other_key other_value '(.|(.))' -s '*print #[key] #{1,key} #[other_key] #{2,key} [#[unknown_key]]'"
echo A | pcresp -d key value -d other_key other_value '(.|(.))' -s '*print #[key] #{1,key} #[other_key] #{2,key} [#[unknown_key]]'
echo

# Syntax error
echo "A | pcresp '.' -s '*print #[key] #a'"
echo A | pcresp '.' -s '*print #[key] #a'
echo
