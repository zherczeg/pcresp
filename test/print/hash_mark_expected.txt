echo A | pcresp '(.)' -s '*print    AB    CD   <AB  CD>    < AB >     CD'
AB CD AB  CD  AB  CD

A | pcresp '(*:MarK)(.)' -s '*print #1 #{1} #< #> ## #M'
A A < > # MarK

A | pcresp -d key value -d other_key other_value '(.|(.))' -s '*print #[key] #{1,key} #[other_key] #{2,key} [#[unknown_key]]'
value A other_value value []

A | pcresp '.' -s '*print #[key] #a'
Cannot compile: *print #[key] #<< SYNTAX ERROR HERE >>a
    Error at offset 15 : invalid # (hash mark) sequence

